src/main/java/io/axoniq/axonserver/connector/impl/HeartbeatMonitor.java

```diff
@@ -102,12 +102,13 @@ public void disableHeartbeat() {
 
     private void checkAndReschedule(int task) {
         if (task != taskId.get()) {
+            // heartbeats should not be considered valid when a change was made
             return;
         }
         long delay = Math.min(interval.get(), 1000);
         try {
-            // heartbeats should not be considered valid when a change was made
-            checkBeat();
+            checkBeatDeadline();
+            sendBeatIfTimeElapsed();
             logger.debug("Heartbeat status checked. Scheduling next heartbeat verification in {}ms", delay);
         } catch (Exception e) {
```

Comment:
Shouldn't we move this catch to the respective "checkBeatDeadline" and "sendBeatIfTimeElapsed" methods? The root cause of an exception in those two methods will be different, which should be reflected in the exception message.

Reply:
I've taken a look of this and I think it's better to leave it as-is. If we have two separate try-catches, a second process will spawn (since both the `finally` block and the normal process reschedule the heartbeat check), leading to a double heartbeat syndrome